### ISSUE TITLE
when in doubt, --fastq_chars should assume an offset of +33 rather than +64

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/vsearch#238]
The offset-ambiguous example below:
```sh
vsearch --fastq_chars <(printf "@s\nAA\n+\nII\n")
```
returns `Guess: Illumina 1.5+ format (phred+64)`.

Given that nucleotides with high quality values are more frequent than nucleotides with very low quality values, and given that the dominating offset has been +33 for several years now, then the `I` symbols in the example should be interpreted as high quality values (40, with an offset of +33) rather than low quality values (9, with an offset of +64). Therefore, the guess should be `+33`.

What do you think?

**Comment by torognes** [comment]
As it is now, VSEARCH will read the entire file and find the lowest and highest quality character used. If that lowest character is between 33 (!) and 58 (:) it chooses the +33 offset. If not, it chooses the +64 offset. If the lowest character is above 58 there will always be doubt.

There is a nice illustration of this here: https://en.wikipedia.org/wiki/FASTQ_format#Encoding

If "I" is the only character, I agree that +33 is the best guess. But I would think that it would be very odd if one of these big files would not contain a single nucleotide with quality below 26 (corresponding to character ":").

Anyway, I see now that USEARCH does not report any guess for the offset anymore (it did earlier). It just reports the range of values used. It is perhaps better to leave the choice to the user when we cannot say anything certain. Perhaps VSEARCH could just report either offset +33 if the lowest quality character is below 59 and report that it is uncertain if the lowest quality character is 59 or higher?

**Comment by frederic-mahe** [comment]
> As it is now, VSEARCH will read the entire file and find the lowest and highest quality character used. If that lowest character is between 33 (!) and 58 (:) it chooses the +33 offset. If not, it chooses the +64 offset. If the lowest character is above 58 there will always be doubt.

> Perhaps VSEARCH could just report either offset +33 if the lowest quality character is below 59 and report that it is uncertain if the lowest quality character is 59 or higher?

I think vsearch should report a guess value, even when there is an ambiguity. After all, that's clearly indicated as a guess, and users can use the qmin-qmax range to decide what to do next.
```sh
Guess: -fastq_qmin 9 -fastq_qmax 9 -fastq_ascii 64
Guess: Illumina 1.5+ format (phred+64)
```

Here is what I suggest (assuming the Q values range from 0 to 41 + offset):
- if the observed range (min and max observed Q values + offset) falls anywhere between 75 and 105, then guess +64.
- if the observed range falls anywhere between 64 and 74 (the overlap between the two offsets), then this is an ambiguous situation. I'd argue that vsearch should guess +33 (see my first post).
- consequently, if the observed range falls anywhere between 33 and 74, then guess +33.
- of course, if Q values are observed outside of the 33-105 range, then this is an error.

**Comment by torognes** [comment]
In commit 38a148e3465a6fa9d1829ca9807ff25ab550a3b9, I've changed the guessing to suggest the +33 offset in more cases. If either the lowest quality score observed is below 59 or the highest quality score observed is below 75, the guess is +33, otherwise it is +64.

I've now seen references to an Illumina 1.9 version as well. It may have offset +33 and quality values as high as 50 for some of the newer machines and chemistries.

Also, consensus sequences assembled from two or more sequences may generate even higher values (e.g. for the `vsearch --fastq_mergepairs` command).

**Comment by torognes** [comment]
Fixed in version 2.12.0 just released.
